benchmarks/perf-tool/okpt/io/config/parsers/util.py

```diff
@@ -6,30 +6,24 @@
 
 """Utility functions for parsing"""
 
-from dataclasses import dataclass
-from typing import Union, cast
-import h5py
 
 from okpt.io.config.parsers.base import ConfigurationError
+from okpt.io.dataset import HDF5DataSet, BigANNNeighborDataSet, \
+    BigANNVectorDataSet, DataSet, Context
 
 
-@dataclass
-class Dataset:
-    train: h5py.Dataset
-    test: h5py.Dataset
-    neighbors: h5py.Dataset
-    distances: h5py.Dataset
+def parse_dataset(dataset_format: str, dataset_path: str,
+                  context: Context) -> DataSet:
+    if dataset_format == 'hdf5':
+        return HDF5DataSet(dataset_path, context)
 
+    if dataset_format == 'bigann' and context == Context.NEIGHBORS:
+        return BigANNNeighborDataSet(dataset_path)
```

Comment:
What is the behavior in case dataset is in some unsupported format or in a wrong context, do we need to validate and maybe throw some error in such case?

Reply:
Behavior is undefined. I will add some validation.